src/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -505,18 +527,29 @@ private HitsForInsert nextAligned() {
 
     /**
      * Copies alignment info from aligned to unaligned read, clips as appropriate, and sets PG ID.
+     * May also un-map the resulting read if the alignment is bad (e.g. no unclipped bases).
```

Comment:
Illegal in what sense? Zero unclipped bases is a condition we check for here (in ```cigarMapsNoBasesToRef```), so I assume it shows up in raw aligner output from time to time.